### PR TITLE
Add an `xtask` package to simplify our CI workflow

### LINKED ISSUE #11: Add support for `xtask` to simplify checks
We can integrate `xtask` support to have a command that checks all the combinations of features for a certain chip.

### LINKED ISSUE #13: CI: Run unit tests
We probably also should run unit tests in CI (now that we have them) but I'm not good at CI stuff

_Originally posted by @bjoernQ in https://github.com/esp-rs/esp-generate/issues/10#issuecomment-2407440778_

### LINKED ISSUE #16: Remove stack-protection
Since its not required for an MVP, and requires nigthly, we will remove it for the time being and use stable for riscv targets

### LINKED ISSUE #21: Generated code leads to runtime panic
Hi!

I have generated a project using `esp-generate --chip esp32c3 -o ble -o embassy -o alloc -o probe-rs -o wokwi -o ci xiao-no-std-esp-gen`

When I run `cargo run` I get the following panic:

```
WARNING: use --release
  We *strongly* recommend using release profile when building esp-hal.
  The dev profile can potentially be one or more orders of magnitude
  slower than release, and may cause issues with timing-senstive
  peripherals and/or devices.

    Finished `dev` profile [optimized + debuginfo] target(s) in 0.25s
     Running `probe-rs run --chip=esp32c3 target\riscv32imc-unknown-none-elf\debug\async_main`
      Erasing ✔ [00:00:01] [######################################################################################################################################] 320.00 KiB/320.00 KiB @ 260.17 KiB/s (eta 0s )
  Programming ✔ [00:00:02] [#######################################################################################################################################] 118.05 KiB/118.05 KiB @ 57.46 KiB/s (eta 0s )    Finished in 3.335s
ERROR probe_rs::util::rtt: 
Error reading from RTT: The control block has been corrupted. write pointer is 780964987 while buffer size is 1024 for up channel 0 (defmt)
INFO  Embassy initialized!
└─ async_main::____embassy_main_task::{async_fn#0} @ src\bin\async_main.rs:24  
ERROR panicked at 'init called more than once!'
└─ esp_hal::soc::implementation::peripherals::{impl#0}::take::{closure#0} @ C:\Users\ckren\.cargo\registry\src\index.crates.io-6f17d22bba15001f\esp-hal-0.21.1\src\fmt.rs:101 
ERROR ====================== PANIC ======================
└─ esp_backtrace::panic_handler @ C:\Users\ckren\.cargo\registry\src\index.crates.io-6f17d22bba15001f\esp-backtrace-0.14.2\src\lib.rs:25  
ERROR panicked at C:\Users\ckren\.cargo\registry\src\index.crates.io-6f17d22bba15001f\defmt-0.3.8\src\lib.rs:367:5:
explicit panic
└─ esp_backtrace::panic_handler @ C:\Users\ckren\.cargo\registry\src\index.crates.io-6f17d22bba15001f\esp-backtrace-0.14.2\src\lib.rs:25
```

Greetings,
Christian

### PR DESCRIPTION
**Pull request by jessebraham**
This turned out to be a bit bigger of a project than anticipated, but this PR should close 4 issues so that's a win at least 😁 

- Added an `--output-path` option so that we can generate the projects in arbitrary directories
    - This is so that the `xtask` can generate projects in temporary directories, to avoid polluting the workspace when run locally
- Removed the `stack-protector` options (closes #16)
- Created the `xtask` package to simplify our CI workflow (closes #11)
- Updated the CI workflow to use `xtask` package and run unit tests (closes #13)
- Removed the extraneous HAL initialization call from `async_main.rs` (should close #21)

The output of the checks are a bit noisy, maybe we want to avoid displaying `stdout` and `stderr` to make the logs a bit more readable?

I've also just copied the combinations of features from the existing workflow, perhaps we want to check more?

Happy to make any changes, lemme know what you guys think.

### CODE REVIEW

**Review by SergioGasquez — APPROVED** [review]
LGTM! Thanks for fixing/improving so many things! 

Current CI takes about 14 mins, we may want to check all the combinations for every chip, but that will increase the CI time. Maybe run it daily instead of running it per commit? This would catch issue like #19 #20

File: README.md [hunk 1]
```diff
@@ -11,23 +11,29 @@ Template generation tool to create `no_std` applications targeting Espressif's l
...
+
+   1. Using TUI:
+
+      ```
+      esp-generate --chip esp32 tests
+      ```
+
+      Replace the chip and project name accordingly and choose the different options using the TUI.
+
+   2. Adding the options to the `esp-generate command:
```
Comment (bjoernQ) [inline README.md:31]: `esp-generate`

**Comment by MabezDev** [comment]
> Current CI takes about 14 mins

Unlike esp-hal, I see this repo getting less and less activity as it stabilizes. Imo CI time is less of a concern here, but I'll leave it with you guys for the final call.

**Comment by bjoernQ** [comment]
> > Current CI takes about 14 mins
> 
> Unlike esp-hal, I see this repo getting less and less activity as it stabilizes. Imo CI time is less of a concern here, but I'll leave it with you guys for the final call.

Wanted to write the same comment - if this assumption turns out to be wrong,, we can reconsider it

**Pull request merged**